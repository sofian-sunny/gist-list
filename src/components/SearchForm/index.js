import React, { useRef } from "react";
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import cn from './search-form.module.scss'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

/**
 * This is SearchForm Method
 * this shows search bar in the Home page
 * @reactProps {string} placeholderText - search input message
 * @reactProps {string} onSubmitSearchForm - callback function
 */
const SearchForm = (props) => {
  const searchInput = useRef()
  const { placeholderText, onSubmitSearchForm } = props

  const onSubmitForm = (e) => {
    e.preventDefault()
    const searchQuery = searchInput.current.value
    !!searchQuery && onSubmitSearchForm(searchInput.current.value)
  }


  return (
    <div className={cn.wrap}>
      <form className={cn.search} onSubmit={(e) => onSubmitForm(e)}>
        <input ref={searchInput} type="text" className={cn.searchTerm} placeholder={placeholderText} />
        <button id="search" type="submit" className={cn.searchButton}>
          <FontAwesomeIcon icon={faSearch} />
        </button>
      </form>
    </div>
  )
}

SearchForm.propTypes = {
  placeholderText: PropTypes.string,
  onSubmitSearchForm: PropTypes.func
}

SearchForm.defaultProps = {
  placeholderText: 'What are you looking for?',
  onSubmitSearchForm: ()=> {}
}

export default SearchForm
