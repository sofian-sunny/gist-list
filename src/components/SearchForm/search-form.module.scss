.search {
    width: 100%;
    position: relative;
    display: flex;
  }
  
  .searchTerm {
    width: 100%;
    border: 1px solid #00B4CC;
    border-right: none;
    height: 40px;
    border-radius: 5px 0 0 5px;
    outline: none;
    font-size: 16pt;
    color: #000000;
    padding-left: 10px;
    &:focus {
      color: #000000;
    }
  }
  
  .searchButton {
    width: 80px;
    height: 44px;
    border: 1px solid #00B4CC;
    background: #00B4CC;
    text-align: center;
    color: #fff;
    border-radius: 0 5px 5px 0;
    cursor: pointer;
    font-size: 20px;
  }
    
  .wrap {
    width: 70%;
    margin-top: 100px;
    margin-left: 50%;
    transform: translate(-50%, -50%);
  }