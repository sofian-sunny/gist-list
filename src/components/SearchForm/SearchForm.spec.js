import React from 'react'
import { shallow } from 'enzyme'
import SearchForm from './index'

const setUp = (props) => {
  const component = shallow(<SearchForm {...props} />)

  return component
}

describe('SearchForm Component render without error', () => {
  const props = {
    title: 'greetings title'
  }
  const wrapper = setUp(props)

  it('SearchForm method should return list item with image', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should update gist data on click", () => {
    let submit = jest.fn();
    let wrapper = shallow(<SearchForm onSubmit={(e) => onSubmitForm(e)}/>);
    wrapper.simulate('onSubmit', { submit });
  })

  it("should update gist data on click", () => {
    const button = wrapper.find('#search');
    expect(button.length).toBe(1)
    button.simulate('submit')
  })
})
