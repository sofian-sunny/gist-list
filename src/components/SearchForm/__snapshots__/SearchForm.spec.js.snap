// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Component render without error SearchForm method should return list item with image 1`] = `ShallowWrapper {}`;
